datacite-findable-doi-handler/src/test/java/no/unit/nva/datacite/handlers/FindableDoiEventHandlerTest.java

```diff
@@ -0,0 +1,164 @@
+            () -> findableDoiHandler.handleRequest(inputStream, outputStream, context));
+
+        assertThat(exception.getMessage(), is(equalTo(FindableDoiEventHandler.PUBLICATION_IS_MISSING_ERROR)));
+    }
+
+    @Test
+    public void handleRequestThrowsIllegalArgumentExceptionOnMissingPublicationId() {
+        InputStream inputStream = IoUtils.inputStreamFromResources(
+            Path.of("doi_request_event_empty_publication_id.json"));
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class,
+            () -> findableDoiHandler.handleRequest(inputStream, outputStream, context));
+
+        assertThat(exception.getMessage(), is(equalTo(FindableDoiEventHandler.PUBLICATION_ID_MISSING_ERROR)));
+    }
+
+    @Test
+    public void handleRequestThrowsIllegalArgumentExceptionOnInvalidDoi() {
+        InputStream inputStream = IoUtils.inputStreamFromResources(
+            Path.of("doi_publication_event_invalid_doi.json"));
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class,
+            () -> findableDoiHandler.handleRequest(inputStream, outputStream, context));
+
+        assertThat(exception.getMessage(), is(equalTo(FindableDoiEventHandler.DOI_IS_MISSING_OR_INVALID_ERROR)));
+        assertThat(exception.getCause().getMessage(),
+            is(equalTo(ImmutableDoi.CANNOT_BUILD_DOI_PROXY_IS_NOT_A_VALID_PROXY)));
+    }
+
+    @Test
+    public void handleRequestThrowsIllegalArgumentExceptionOnEmptyDoi() {
+        InputStream inputStream = IoUtils.inputStreamFromResources(
+            Path.of("doi_publication_event_empty_doi.json"));
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class,
+            () -> findableDoiHandler.handleRequest(inputStream, outputStream, context));
+
+        assertThat(exception.getMessage(), is(equalTo(FindableDoiEventHandler.DOI_IS_MISSING_OR_INVALID_ERROR)));
+        assertThat(exception.getCause().getMessage(),
+            is(equalTo(FindableDoiEventHandler.DOI_IS_MISSING_OR_INVALID_ERROR)));
+    }
+
+    @Test
+    void handleRequestThrowsExceptionWhenInputPublicationIdIsInvalid() {
+        InputStream inputStream = IoUtils.inputStreamFromResources(PUBLICATION_EVENT_INVALID_PUBLICATION_ID);
+
+        var actualException = assertThrows(IllegalArgumentException.class,
+            () -> findableDoiHandler.handleRequest(inputStream, outputStream, context));
+        assertThat(actualException.getMessage(),
+            is(equalTo(LandingPageUtil.ERROR_PUBLICATION_LANDING_PAGE_COULD_NOT_BE_CONSTRUCTED)));
+    }
+
+    private String verifyPartsOfMetadata() {
+        return and(
+            contains("JOURNAL_ARTICLE"),
+            and(
+                contains("<title>Conformality loss"),
+                contains(
+                    "identifierType=\"URL\">https://example.net/unittest/namespace/publication/654321</identifier>")));
+    }
```

Comment:
If we want more detailed assertions, we probably want to do this in the `Transformer` itself.